### PR TITLE
Port from Jenkins to Github Actions

### PR DESCRIPTION
**Pull request by BryanQuigley**
Most items copied over with only minor changes, except:

Commenting out update_batch_definitions as that would need a good number of changes
to support not using an AWS profile in Github and we are looking into improving that
as my potential next task anyway.

Slack notifications for build success or failure are covered when pushing a PR
by the Github integration.

REMINDER: Add infra apply when PR approved

Resolves: #882

### Demo

On PR, Terraform is not run: https://github.com/azavea/pfb-network-connectivity/runs/7904131095?check_suite_focus=true
On Push to to test (and others) it is run; https://github.com/azavea/pfb-network-connectivity/runs/7888427429?check_suite_focus=true

## Testing Instructions

 * Confirm GH Action output looks like Jenkins

## Checklist

- [ ] Add entry to CHANGELOG.md

### CODE REVIEW

**Comment by BryanQuigley** [comment]
I left infra apply off until review is complete - plan to add it when it's done.

**Comment by BryanQuigley** [comment]
The failure was due to GH Actions being down.

**Comment by jacobscarfmerrell** [comment]
where in the [push to test action run](https://github.com/azavea/pfb-network-connectivity/runs/7888427429?check_suite_focus=true) from the test instructions does terraform run? I was looking for that `infra` script call that gets skipped in the PR run

**Comment by BryanQuigley** [comment]
Look at this one: https://github.com/azavea/pfb-network-connectivity/actions/runs/3003482711/jobs/4821824010

The ones that are made from the PR will not run it.

**Review by jacobscarfmerrell — APPROVED** [review]
lgtm!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - develop
+      - test/**
```
Comment (KlaasH) [inline .github/workflows/ci.yml:7]: This should include `release/` branches as well, and they should be included in the conditions controlling whether to deploy.  We don't always push release branches when we're doing a release for this project, but if we did we would want them to build.

File: scripts/infra [hunk 2]
```diff
@@ -55,7 +55,7 @@ if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
 
         case "${1}" in
             plan)
-                update_batch_definitions
+                #update_batch_definitions
```
Comment (KlaasH) [inline scripts/infra:58]: I ran through the steps of a staging deploy through `infra play` and got "BATCH_ANALYSIS_JOB_NAME_REVISION: unbound variable", but I guess that's expected for now, and the workaround is just to comment this back in when doing a manual deploy (e.g. to production).

File: .github/workflows/ci.yml (hunk 1 above)
Comment (BryanQuigley) [reply .github/workflows/ci.yml:7]: adding.

**Comment by BryanQuigley** [comment]
We do get build notifications when the build is from a PR.  but not automagically when pushing manually, etc.  That should be just 1 point to add in the future if we find ourselves really needing it.

**Pull request merged**